### DISCUSSION TITLE
How can I invalidate/reset query using QueryClient ?

### DISCUSSION THREAD

**Discussion by WingsDevelopment** [wevm/wagmi#3260] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

1.3.10

### Current Behavior

I am trying to invalidate or reset query using scopeKey: 'somestring' in useContractRead
And in other component I am trying to
 
```
  const queryCache = useQueryClient();
  <Button
        onClick={() => {
          queryCache.invalidateQueries({
            queryKey: ['somestring'],
          });
        }}
      >
        reset query
      </Button>
```

But it doesn't work. why? Am 

### Expected Behavior

To refetch data

**Comment by KholdStare** [comment]
@tmm This was converted from an issue, but never answered? Is there a way to invalidate a "group" of reads/simulate calls?

There might be several places that depend on the current blockchain state, and e.g. I know for a fact that when a particular write succeeds, all those read/simulate calls are invalid now. I want to explicitly invalidate that group and force re-querying the blockchain. The other options are more crude, like `watch` or depending on `blockNumber` which is too many useless/redundant calls when nothing is happening.

**Reply by tmm** [reply]
https://github.com/wevm/wagmi/discussions/4240#discussioncomment-10549269

Please don't @ mention me in the future unless there is a critical issue. The answers for most questions are out there if you look!
